pkg/oc/admin/policy/modify_roles.go

```diff
@@ -513,6 +513,12 @@ func (o *RoleModificationOptions) RemoveRole() error {
 		if err != nil {
 			return err
 		}
+		// Check that we update the rolebinding for the intended role.
+		if existingRoleBinding.RoleRef.Name != o.RoleName || existingRoleBinding.RoleRef.Namespace != o.RoleNamespace {
+			return fmt.Errorf("rolebinding %s contains role %s in namespace %s, instead of role %s in namespace %s",
```

Comment:
Add a test that fails when this case is hit, otherwise LGTM.

Reply:
Please open a separate issuse with all the nits in the original code that you'd like fixed.
I want this PR to unblock QA for 3.9, if time will permit I'll bring down tech debt in a separate PR that closes your issue